### PR TITLE
Sc 270 backend endpoint for create a task

### PR DESCRIPTION
**Pull request by marialobillo**
## Description

As a user I want to be able to create a task
#270

## How has this been tested?

Unit test have been added to the cover the changes.

## My PR contains…

- [ ]  No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ]  Dependency changes (any modification to dependencies in `package.json`)
- [ ]  Bug fixes (non-breaking change which fixes an issue)
- [ ]  Improvements (misc. changes to existing features)
- [x]  Features (non-breaking change which adds functionality)

## My changes…

- [ ]  are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ]  are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ]  are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [ ]  are not breaking changes.

## Documentation

- [ ]  My changes do not require a change to the project documentation.
- [x]  My changes require a change to the project documentation.
- [ ]  If yes to above: I have updated the documentation accordingly.

# Automated tests

- [ ]  My changes can not or do not need to be tested.
- [ ]  My changes can and should be tested by unit and/or integration tests.
- [ ]  If yes to above: I have added tests to cover my changes.
- [ ]  If yes to above: I have taken care to cover edge cases in my tests.
- [x]  All new and existing tests passed.

### CODE REVIEW

File: server/src/app/controllers/taskController.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Request, Response } from 'express'
```
Comment (AbdelrahmanRadwan) [inline server/src/app/controllers/taskController.ts]: This should be part of a presentation layer. Instead of `controller` you can call it `presentation` because a controller is something very specific, but a presentation is a layer with more stuff.

Comment (AbdelrahmanRadwan) [reply server/src/app/controllers/taskController.ts]: It also shouldn't be called `app`. remove the word `app` from the directory.

File: server/src/domain/usecases/createTask.test.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { TaskRepository } from '../../infrastructure/persistance/taskRepository';
```
Comment (AbdelrahmanRadwan) [inline server/src/domain/usecases/createTask.test.ts]: The tests should be in a separate directory next to `src`.

File: server/src/domain/usecases/createTask.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Task, TaskProps } from '../entities/task'
```
Comment (AbdelrahmanRadwan) [inline server/src/domain/usecases/createTask.ts]: I guess this name is a bad name. CreateTask implies that there is logic and action. However, a domain is just data store, no logic.

File: server/src/domain/usecases/createTask.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { TaskRepository } from '../../infrastructure/persistance/taskRepository'
+
+export class CreateTask {
```
Comment (AbdelrahmanRadwan) [inline server/src/domain/usecases/createTask.ts]: the task repository here is storage layer. A storage layer is a layer in the infrastructure.

File: server/src/app/controllers/taskController.ts (hunk 1 above)
Comment (marialobillo) [reply server/src/app/controllers/taskController.ts]: Thanks, I have remove it

File: server/src/domain/usecases/createTask.test.ts (hunk 2 above)
Comment (marialobillo) [reply server/src/domain/usecases/createTask.test.ts]: I put the test directory outside src folder

File: server/src/domain/usecases/createTask.ts (hunk 4 above)
Comment (marialobillo) [reply server/src/domain/usecases/createTask.ts]: Thanks, not sure if I did it well. I put use-cases in application layer

File: server/src/domain/usecases/createTask.ts (hunk 3 above)
Comment (marialobillo) [reply server/src/domain/usecases/createTask.ts]: I relocated use-cases in application layer, and renamed the createTask.ts to createTaskUseCase.ts

File: server/tests/domain/entities/task.test.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    expect(task.media).toBe(taskData.media)
+    expect(task.tags).toBe(taskData.tags)
+  })
+
+  it('should create a new task with a random id', () => {
+    const taskData = {
+      title: 'Task title',
+      description: 'Task description',
+      location: 'Task location',
+      price: 100,
```
Comment (AbdelrahmanRadwan) [inline server/tests/domain/entities/task.test.ts]: These tests don't really test much. Normally you don't add tests to the models because they don't have any logic.

**Review by AbdelrahmanRadwan — APPROVED** [review]

File: server/tests/domain/entities/task.test.ts (hunk 5 above)
Comment (marialobillo) [reply server/tests/domain/entities/task.test.ts]: Ok, thanks, I will remove it
